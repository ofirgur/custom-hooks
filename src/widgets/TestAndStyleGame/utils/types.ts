export interface GameStateContextInterface {
  gameState: GameState;
  setGameState: (gameState: Partial<GameState>) => void;
}

export interface GameState {
  gameScreen: GameScreen;
  gamePosts: GamePost[];
  gameSelectedPost?: GamePost;
}

export enum GameScreen {
  Start = 1,
  Card = 2,
  Over = 3,
}

export interface GamePost {
  userId: number,
  id: number,
  title: string,
  body: string,
}
