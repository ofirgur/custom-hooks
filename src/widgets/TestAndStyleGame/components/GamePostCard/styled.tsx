import styled from '@emotion/styled';

export const GamePostCardStyled = styled.section`
    background: #FFF;
    height: 200px;
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
    padding: 10px;
    margin: 10px;
`;