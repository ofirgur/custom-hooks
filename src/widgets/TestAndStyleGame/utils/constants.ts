import { GameState, GameScreen } from './types';

export const INITIAL_GAME_STATE: GameState = {
  gameScreen: GameScreen.game,
};