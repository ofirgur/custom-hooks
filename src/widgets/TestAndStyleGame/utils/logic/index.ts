export const _sum = (x: number, y: number) => x + y;
export const _toLowwerCase = (str: string) => str.toLocaleLowerCase(); 
export const _reduceToSum = (arr: number[]) => {
    if(!arr) return;
    return arr.reduce((acc: number, item: number) => {
        return acc += item;
    }, 0);
};

export const _sumWithInitial = (initialValue: number, arr: number[]) => arr.reduce(
    (previousValue, currentValue) => previousValue + currentValue,
    initialValue
);