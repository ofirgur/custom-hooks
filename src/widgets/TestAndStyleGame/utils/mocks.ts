
import { GameScreen } from './types';

export const mockedGameState = {
  gameScreen: GameScreen.game,
};