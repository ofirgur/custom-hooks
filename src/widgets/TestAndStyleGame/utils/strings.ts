export const STRINGS = {
  CARD_SCREEN: {
    BACK: 'BACK',
  },
};
