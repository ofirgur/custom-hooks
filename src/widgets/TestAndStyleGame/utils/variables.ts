export const VARIABLES = {
  BACKGROUND: {
    WHITE: '#FFF',
    BLACK: {
      VIEW_PORT: '#16191D',
      BUTTON: '#0E1013',
    },
    ORANGE: {
      SOLID: '#FB8948',
    },
    GRAY: '#C4C4C4',
    CREAME: '#F2D4AD',
    LAND: '#C34C23',
    DUST: '#F2D4AD',
    BEZ: '#F4D0A1',
  },
  COLOR: {
    WHITE: '#FFF',
    ORANGE: {
      SOLID: '#FB8948',
    },
    BLACK: {
      SOLID: '#000',
    },
  },
  BOX_SHADOW: {
    BUTTON: 'inset 2px 1px 10px rgba(0, 0, 0, 0.8)',
  },
  FONT_FAMILY: {
    POPPINS: 'Poppins',
  },
  FONT_STYLE: {
    NORMAL: 'font-style',
  },
  FONT_WEIGHT: {
    NORMAL: '300',
  },
  BREAK_POINTS: {
    LT_MIN: '400px',
    MD_MIN: '690px',
  },
};
