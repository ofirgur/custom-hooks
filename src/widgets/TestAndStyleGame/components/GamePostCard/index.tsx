import React from 'react';

import { GameScreen, GamePost } from '../../utils/types'; 
import { useGameStateContext } from '../../utils/context';
import  { GamePostCardStyled } from './styled';

interface GamePostCardProps {
  post: GamePost;
}

const GamePostCard = (props: GamePostCardProps) => {
  const { post } = props;
  const { setGameState } = useGameStateContext();

  const handleClick = () => {
    setGameState({
      gameScreen: GameScreen.Card,
      gameSelectedPost: post,
    });
  };
  
  return (
    <GamePostCardStyled onClick={handleClick}>
      {post.title}
    </GamePostCardStyled>
  );
};

export default GamePostCard;
