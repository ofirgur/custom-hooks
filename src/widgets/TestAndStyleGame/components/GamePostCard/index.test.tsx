import * as React from 'react';
import { render } from '@testing-library/react';

import { mockedGameState } from '../../utils/mocks';
import GamePostCard from './index';

const mockedSetGameState = jest.fn();
jest.mock('../../utils/context', () => {
  return {
    useGameStateContext: () => ({
      setGameState: mockedSetGameState,
      gameState: {
        ...mockedGameState,
      },
    }),
  }
});

const props = {
    post: mockedGameState.gameSelectedPost,
}

describe('GamePostCard', () => {
  it('should render correctly', () => {
    const { container } = render(<GamePostCard {...props} />);
    expect(container).toMatchSnapshot();
  });
});
