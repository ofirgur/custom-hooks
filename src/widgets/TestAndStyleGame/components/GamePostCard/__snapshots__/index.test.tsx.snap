// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamePostCard should render correctly 1`] = `
<div>
  <section
    class="css-wi1r3z"
    role="button"
  >
    ea molestias quasi exercitationem repellat qui ipsa sit aut
  </section>
</div>
`;
